authority/options.go

```diff
@@ -266,6 +266,14 @@ func WithAdminDB(d admin.DB) Option {
 	}
 }
 
+// WithProvisioners is an option to set the provisioner collection.
+func WithProvisioners(ps *provisioner.Collection) Option {
```

Comment:
Can we mark this function as deprecated and add a nolint in the cli?

```suggestion
// WithProvisioners is an option to set the provisioner collection.
// 
// Deprecated: provisioner collections will likely change
func WithProvisioners(ps *provisioner.Collection) Option {
```